src/test/java/dmytro/hadiuchko/springboot/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,118 @@
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+    private static final String CATEGORY_DESCRIPTION = "interesting book";
+    private static final String CATEGORY_NAME = "Fiction";
+    private static final Long CATEGORY_ID = 1L;
+    private static final String CATEGORY_URL = "/api/categories";
+    private static final String CATEGORY_BY_ID_URL = "/api/categories/1";
+    private static final String A_CATEGORY_FOR_FICTIONAL_BOOKS = "A category for fictional books.";
+
+    private static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext webApplicationContext) {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    @DisplayName("Create new category")
+    void createCategory_validAdminRole_success() throws Exception {
+        CategoryRequestDto requestDto = new CategoryRequestDto(CATEGORY_NAME, CATEGORY_DESCRIPTION);
+
+        CategoryResponseDto expected = new CategoryResponseDto();
+        expected.setId(CATEGORY_ID);
+        expected.setName(requestDto.name());
+        expected.setDescription(requestDto.description());
+
+        String jsonRequest = objectMapper.writeValueAsString(expected);
+
+        MvcResult result = mockMvc.perform(post(CATEGORY_URL)
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+        CategoryResponseDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                CategoryResponseDto.class);
+
+        assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @Sql(scripts = "classpath:sql/scripts/insert-into-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:sql/scripts/drop-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @WithMockUser(username = "user", roles = {"ADMIN", "USER"})
+    @DisplayName("Return all categories")
+    void getAllCategories_withUserRole_ReturnAll() throws Exception {
+        mockMvc.perform(get(CATEGORY_URL))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.length()").value(5))
+                .andExpect(jsonPath("$[0].name").value(CATEGORY_NAME))
+                .andExpect(jsonPath("$[0].description").value(A_CATEGORY_FOR_FICTIONAL_BOOKS));
```

Comment:
Please be consistent and use one type of object check using objectMapper as you did before, or like in this method. I prefer objectmapper